### PR TITLE
Add splinter repository design

### PR DESCRIPTION
**Pull request by peterschwarz**
This change adds the initial proposal for the splinter artifact  repository layout and tools. It covers the requirements for saplings and smart contracts, as well as introduces the concept of a Capability.

A Capability is a distribution of other artifacts that will make up a complete application.  It defines a set of dependencies that include smart contracts and saplings, though may include additional plugin components, such as state delta export processors and CLI tools.

[Rendered](https://github.com/peterschwarz/splinter-docs/blob/repository-design/community/planning/capabilities-repository.md)

### CODE REVIEW

**Comment by vaporos** [comment]
Looks like it needs a link from the features page.

File: community/planning/capabilities-repository.md [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+things such names or dependency relationships.
+
+For example, the Grid Product, version 1.0, smart contract would be stored at
+
+`/contracts/grid_product/grid_product-1.0.scar`
+
+### Tools
+
+Interacting with the capabilities repository requires a set of tools.  For
+developers, these  for creating artifacts and their manifests, as well as
```
Comment (ltseeley) [inline community/planning/capabilities-repository.md:80]: Seems like a word is missing: "these  for", maybe "these are for"?

File: community/planning/capabilities-repository.md [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+publishing artifacts to the repository.  For repository maintainers, tools
+include removing artifacts and generating indexes files of the repository.
```
Comment (ltseeley) [inline community/planning/capabilities-repository.md:82]: s/indexes/index/

File: community/planning/capabilities-repository.md [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+
+## Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+### Manifest Formats
+
+All the major components in the repository have a manifest describing their
+name, and version, as well as optional fields like description.
```
Comment (ltseeley) [inline community/planning/capabilities-repository.md:90]: s/name,/name/

File: community/planning/capabilities-repository.md [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+
+#### Saplings
+The sapling manifest is a JSON file, as this is the most readily consumable
+format by the browsers
```
Comment (ltseeley) [inline community/planning/capabilities-repository.md:94]: s/by/for/, or "...the format most readily consumable by web browsers"

File: community/planning/capabilities-repository.md [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+
+The `./preview` directory contains an optional set of screenshots of the
+sapling.
+
+The file is compressed using the `bzip2` compression algorithm.
+
+#### Smart Contracts
+
+Smart contract packaging follows the SCAR file format defined by Sawtooth Sabre.
+#### Capabilities
```
Comment (ltseeley) [inline community/planning/capabilities-repository.md:203]: Missing a blank line here

File: community/planning/capabilities-repository.md [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+##### Packaging
+
+The packaging tools will be supplied in the project template's `package.json`,
+via `devDependencies`.  These scripts will build the bzip2 package for
+publishing.  The user will be able to run
+
+```
+$ npm run sapling:package
+```
+#### Smart Contracts
```
Comment (ltseeley) [inline community/planning/capabilities-repository.md:311]: Missing blank line

File: community/planning/capabilities-repository.md [hunk 7]
```diff
@@ -0,0 +1,413 @@
...
+  and [indexes](https://maven.apache.org/repository/central-index.html)
+* [Crates.io](https://github.com/rust-lang/crates.io)
+
+## Unresolved questions
+[unresolved]: #unresolved-questions
+
+This design does not cover any advanced search or indexing features, such as
+tags or content analysis.
+
+Likewise, it assumes all publish will currently be handled by the repository
```
Comment (ltseeley) [inline community/planning/capabilities-repository.md:412]: s/publish/publishing/

File: community/planning/capabilities-repository.md (hunk 4 above)
Comment (peterschwarz) [reply community/planning/capabilities-repository.md]: Reworded as suggested

File: community/planning/capabilities-repository.md (hunk 5 above)
Comment (peterschwarz) [reply community/planning/capabilities-repository.md:204]: Fixed

File: community/planning/capabilities-repository.md (hunk 6 above)
Comment (peterschwarz) [reply community/planning/capabilities-repository.md:313]: Fixed

**Review by ltseeley — APPROVED** [review]

**Review by rbuysse — APPROVED** [review]

**Pull request merged**